Tribler/Core/Modules/resource_monitor.py

```diff
@@ -33,8 +44,52 @@ def start(self):
             self.session.config.get_resource_monitor_poll_interval(), now=False)
 
     def stop(self):
+        if HAS_YAPPI and self.profiler_running:
+            self.stop_profiler()
+
         self.shutdown_task_manager()
 
+    def start_profiler(self):
+        """
+        Start the Yappi profiler if the library is available and if it's not already running.
+        """
+        if self.profiler_running:
+            raise RuntimeError("Profiler is already running")
+
+        if not HAS_YAPPI:
+            raise RuntimeError("Yappi is not installed")
+
+        yappi.start(builtins=True)
+        self.profiler_start_time = int(time.time())
+        self.profiler_running = True
+
+    def stop_profiler(self):
+        """
+        Stop yappi and write the stats to the output directory.
+        Return the path of the yappi statistics file.
+        """
+        if not self.profiler_running:
+            raise RuntimeError("Profiler is not running")
+
+        if not HAS_YAPPI:
+            raise RuntimeError("Yappi is not installed")
```

Comment:
It would be nice to add more comments on this runtime error. Considering user is seeing this error, it might be a good idea to point how to install yappi and that Tribler needs to be restarted after yappi installation to enable it.